### PR TITLE
USB-Service added

### PR DESCRIPTION
**Pull request by leogaube**
Wow, it is truly amazing how easy you made it for anyone to add new service providers with no need to integrate anything manually. A single file with "svc_" in the filename is enough!

This service adds USB-stick support (or support for any other external storage device)!
I think this new service could be really useful for someone with a bad/metered internet connection.
Or maybe a friend just came back from a vacation and wants to show you some of his/her pictures.

**Here is how it works:**
You need to create a "/photoframe" directory on your storage device and place any "albums" (directories) inside. Just plug in your USB-stick / hard drive / SD-card (via USB-card reader), add a USB-Service (web interface), and add any album name as keywords (case-sensitive!!). If you want to display all albums just use "ALLALBUMS" as keyword. 
Alternatively, you can just place images directly inside the "/photoframe" folder and it will detect those pictures automatically without any keywords!
By the way, photoframe automatically detects storage devices and will try to mount the most recent one to "/mnt/usb". It should also detect when you unplug or replug a device.

**NOTE:** ~~Currently only a single storage device is supported, so if you add another service it will use the same device for now. However, I will add multi-USB-stick support in the future.~~
Of course photoframe still detects if you switch to a different USB-stick and it will delete invalid keywords automatically. If you are quick, you can even unplug the storage device, add some more images, plug it back into the photoframe and the slideshow will just continue without any interruptions.

Unfortunately, the USB-Service only seems to work with the changes of my previous Pull Request, so please make sure to merge the other one first!

### CODE REVIEW

**Comment by leogaube** [comment]
I adapted the service to use the general approach from my changes to the previous pull request. 
Of course a usb-service does not have to "download" an image but I used `requestUrl` nonetheless:
https://github.com/mrworf/photoframe/blob/6fafd1356e1ce8bcbdbae33ba515dc08c7e31b7c/services/svc_usb.py#L249-L251

I hope the naming and behaviour of this function does not bother you, but I figured it would be the easiest way to make it work with the `selectImageFromAlbum` function of `base.py`

**Comment by leogaube** [comment]
I also added support for multiple usb-sticks:
Just create as many USB-Services as you have storage devices connected and each service will automatically mount one usb-device.

File: services/svc_usb.py [hunk 1]
```diff
@@ -91,21 +94,32 @@ def checkForInvalidKeywords(self):
...
-    if not os.path.exists(USB_Photos.BASE_DIR):
+    state = None
+    if not os.path.exists(self.baseDir):
       if not self.mountStorageDevice():
-        return BaseService.STATE_NOT_CONNECTED
-    if len(self.getAllAlbumNames()) == 0 and len(self.getBaseDirImages()) == 0:
-      self.unmountBaseDir()
-      return BaseService.STATE_NO_IMAGES
-    return BaseService.updateState(self)
+        state = BaseService.STATE_NOT_CONNECTED
```
Comment (mrworf) [inline services/svc_usb.py]: Should not use/create a new state in base service, needs to be stored locally. Ideally you simply use `STATE_NO_IMAGES` but have a local substate indicating why which you can use in `explainState()` below.

File: services/svc_usb.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    return {'error': None, 'keywords': keyword, 'extras': None}
+
+  def getKeywords(self):
+    if not os.path.exists(USB_Photos.BASE_DIR):
+      #wrong error TODO
+      return []
+
+    keywords = list(self._STATE['_KEYWORDS'])
+    if "ALLALBUMS" in keywords:
+      # No, you can't have an album called /photoframe/ALLALBUMS ...
```
[the remark this answers is not in the document]
Comment (leogaube) [reply services/svc_usb.py]: ok :)

File: services/svc_usb.py (hunk 1 above)
Comment (leogaube) [reply services/svc_usb.py]: i changed it

**Comment by mrworf** [comment]
Looks good, Once #84 is merged, I'll pull this one in as well.

**Pull request merged**